tests/vm_test.go

```diff
@@ -1323,61 +1320,67 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 					Expect(err).ToNot(HaveOccurred())
 
-					By("Starting the VMI with virtctl")
-					startCommand := clientcmd.NewRepeatableVirtctlCommand(virtctl.COMMAND_START, "--namespace", vm.Namespace, vm.Name)
-					Expect(startCommand()).To(Succeed())
-
-					By("Waiting for VM to be in Starting status")
-					Eventually(ThisVM(vm), 160*time.Second, 1*time.Second).Should(HavePrintableStatus(v1.VirtualMachineStatusStarting))
+					By("waiting for virt-launcher")
+					var vmi *v1.VirtualMachineInstance
+					Eventually(func() bool {
+						vmi, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+						if errors.IsNotFound(err) {
+							return false
+						}
+						Expect(err).ToNot(HaveOccurred())
+						_, err = libpod.GetPodByVirtualMachineInstance(vmi, vm.Namespace)
+						if errors.IsNotFound(err) {
+							return false
+						}
+						Expect(err).ToNot(HaveOccurred())
+						return true
+					}).WithTimeout(60*time.Second).WithPolling(5*time.Second).Should(BeTrue(), "virt-launcher should be created eventually")
 
-					By("Waiting for VMI to fail")
-					Eventually(ThisVMIWith(vm.Namespace, vm.Name), 480*time.Second, 1*time.Second).Should(BeInPhase(v1.Failed))
+					getComputeContainerStateRunning := func() (*k8sv1.ContainerStateRunning, error) {
+						launcherPod, err := libpod.GetPodByVirtualMachineInstance(vmi, vm.Namespace)
+						if err != nil {
+							return nil, err
+						}
+						for _, status := range launcherPod.Status.ContainerStatuses {
+							if status.Name == "compute" {
+								return status.State.Running, nil
+							}
+						}
+						return nil, nil
+					}
 
 					// If the annotation v1.KeepLauncherAfterFailureAnnotation is set to true, the containerStatus of the
 					// compute container of the virt-launcher pod is kept in the running state.
 					// If the annotation v1.KeepLauncherAfterFailureAnnotation is set to false or not set, the virt-launcher pod will become failed.
 					By("Verify that the virt-launcher pod or its container is in the expected state")
-					vmi, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
-					Expect(err).ToNot(HaveOccurred())
-					launcherPod, err := libpod.GetPodByVirtualMachineInstance(vmi, vm.Namespace)
-					Expect(err).ToNot(HaveOccurred())
-
 					if toKeep, _ := strconv.ParseBool(keepLauncher); toKeep {
-						Consistently(func() bool {
-							for _, status := range launcherPod.Status.ContainerStatuses {
-								if status.Name == "compute" && status.State.Running != nil {
-									return true
-								}
-							}
-							return false
-						}, 10*time.Second, 1*time.Second).Should(BeTrue())
+						Consistently(func() *k8sv1.ContainerStateRunning {
+							running, err := getComputeContainerStateRunning()
+							Expect(err).ToNot(HaveOccurred())
+							return running
+						}).WithTimeout(10 * time.Second).WithPolling(1 * time.Second).ShouldNot(BeNil())
```

Comment:
We want to check it is true as well

Reply:
I'm not sure I understand what should be true as well. Can you please elaborate?

I changed the name 'running' to something else to avoid confusion, as this is not a boolean.

Reply:
>I changed the name 'running' to something else to avoid confusion, as this is not a boolean.

That will be it